src/main/java/buildcraftAdditions/proxy/ClientProxy.java

```diff
@@ -59,4 +62,8 @@ public EntityPlayer getPlayer(String name) {
 	public void cloneFluidTextures(Fluid source, Fluid target) {
 		target.setIcons(source.getStillIcon(), source.getFlowingIcon());
 	}
+
+	public static GuiScreen getCurrentOpenedScreen() {
+		return Minecraft.getMinecraft().currentScreen;
```

Comment:
also I found that the Gui base class has it's own checker for shift clicking so I just started using that. https://github.com/BigXplosion/Buildcraft-Additions/commit/ddcb66cc6b38dd44c29486167262fbd4f9de0281#diff-4bb8185d9a37917570ff27cfc825bb3fR188